server/src/main/java/org/opensearch/index/SegmentReplicationPressureService.java

```diff
@@ -154,4 +182,95 @@ public void setMaxAllowedStaleReplicas(double maxAllowedStaleReplicas) {
     public void setMaxReplicationTime(TimeValue maxReplicationTime) {
         this.maxReplicationTime = maxReplicationTime;
     }
+
+    @Override
+    public void close() throws IOException {
```

Comment:
Where does this newly added method get called?

Reply:
Currently, it is not called directly from anywhere. I took the reference from [PersistentTasksClusterService](https://github.com/opensearch-project/OpenSearch/blob/4511354122f8b423bbb78d95b7ed6d699450a9b1/server/src/main/java/org/opensearch/persistent/PersistentTasksClusterService.java#L126), just to make sure when service is closed this async task is also closed.